### PR TITLE
Fix 1121 broken use readme test

### LINKED ISSUE #1121: [FR] Improved fix for broken test `test-use_readme.R`
Via https://github.com/ThinkR-open/golem/commit/9ad97753a585ca1ec418025f3ff08db629592702 a test fix for `test-use_readme()` is done that skips the test in non-interactive mode.

However, when eliminating the `devtools:::build_readme()` call at the very end, the test passes successfully in non-interactive mode, so no need for skipping.

Hence, to improve codecov one may delete the `devtools:::build_readme()` line and run the test as usual.

Can be fixed via #1122

### PR DESCRIPTION
**Pull request by ilyaZar**
Fix #1121 

Make all test for `test-use_readme.R` pass successfully in non-interactive mode.

Additionally, test a previously untested corner case.

Improves codecov for `use_readme.R` from 34.29% to 100%

### CODE REVIEW

File: tests/testthat/test-use_readme.R [hunk 1]
```diff
@@ -23,11 +23,16 @@ test_that("check_overwrite works", {
     check_overwrite(FALSE, golem_sys("utils/empty_readme.Rmd")),
     "README.Rmd already exists. Set `overwrite = TRUE` to overwrite."
   )
+  # Check if file exists
+  expect_true(file.exists(golem_sys("utils/empty_readme.Rmd")))
+  # Remove file via check_overwrite setting overwrite=TRUE
+  check_overwrite(TRUE, golem_sys("utils/empty_readme.Rmd"))
```
Comment (ColinFay) [inline tests/testthat/test-use_readme.R:29]: Issue here is that the file will be removed from the installed package on the machine, so I'll rewrite this a bit :)

**Comment by ColinFay** [comment]
Cherry picked in https://github.com/ThinkR-open/golem/pull/1131
